Automattic/wp-calypso
# Components: Remove unused SiteTitle component

#### Changes proposed in this Pull Request

The `SiteTitle` component has been unused since we removed the good old Calypso version of the Jetpack Onboarding plugin in #38000. This PR removes that component as well.

#### Testing instructions

* Verify that the removed component is not used anywhere.
* Verify that `/devdocs/design` still loads well.
